eng/Generation.psm1

```diff
@@ -50,8 +50,12 @@ function AutoRest-Reset()
     Invoke "$script:autoRestBinary --reset"
 }
 
-function Invoke-Cadl($baseOutput, $projectName, $sharedSource="", $fast="", $debug="")
+function Invoke-Cadl($baseOutput, $projectName, $arguments, $sharedSource="", $fast="", $debug="")
```

Comment:
By default the cadl file name is the same as projectName, if not we can add another optional parameter '$cadlFile'. If the $cadlFile is not provide, use the $projectName. It is not runtime $arguments

Reply:
We will add prefix/suffix to project name, so the cadl folder name will not be the same as project name. I would change $arguments to $mainFile